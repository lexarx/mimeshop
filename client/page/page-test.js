jest.mock('products-list/products-list', () => () => 'ProductList');
jest.mock('product-page/product-page', () => () => 'ProductPage');
jest.mock('cart-page/cart-page', () => () => 'CartPage');
import React from 'react'
import Page from 'page/page'
import { render } from 'enzyme'

test('displays products list', () => {
	history.pushState(undefined, undefined, '/');
	const page = render(<Page/>);
	expect(page).toMatchSnapshot();
});

test('displays products list', () => {
	history.pushState(undefined, undefined, '/products');
	const page = render(<Page/>);
	expect(page).toMatchSnapshot();
});

test('displays product page', () => {
	history.pushState(undefined, undefined, '/products/0');
	const page = render(<Page/>);
	expect(page).toMatchSnapshot();
});

test('displays cart page', () => {
	history.pushState(undefined, undefined, '/cart');
	const page = render(<Page/>);
	expect(page).toMatchSnapshot();
});