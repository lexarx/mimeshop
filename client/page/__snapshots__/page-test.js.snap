// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays cart page 1`] = `
Array [
  <div
    class="header"
  >
    <span
      class="headerLink"
    >
      Products
    </span>
    <span
      class="headerLink currentHeaderLink"
    >
      Cart
    </span>
  </div>,
  "CartPage",
]
`;

exports[`displays product page 1`] = `
Array [
  <div
    class="header"
  >
    <span
      class="headerLink"
    >
      Products
    </span>
    <span
      class="headerLink"
    >
      Cart
    </span>
  </div>,
  "ProductPage",
]
`;

exports[`displays products list 1`] = `
Array [
  <div
    class="header"
  >
    <span
      class="headerLink currentHeaderLink"
    >
      Products
    </span>
    <span
      class="headerLink"
    >
      Cart
    </span>
  </div>,
  "ProductList",
]
`;

exports[`displays products list 2`] = `
Array [
  <div
    class="header"
  >
    <span
      class="headerLink currentHeaderLink"
    >
      Products
    </span>
    <span
      class="headerLink"
    >
      Cart
    </span>
  </div>,
  "ProductList",
]
`;
